### PR TITLE
Kinds: Add pluralName override for Preferences

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This adds a `pluralName` for `Preferences` to avoid the appending of an extra `s`

**Why do we need this feature?**

This is needed for #69115 

**Who is this feature for?**

@grafana/grafana-app-platform-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: kinds/preference/preference_kind.cue [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package kind
 
-name:        "Preferences"
-maturity:    "merged"
-description: "The user or team frontend preferences"
+name:              "Preference"
```
Comment (toddtreece) [inline kinds/preference/preference_kind.cue:3]: the biggest change is related to preferences. so i had to change the name and parent folder from plural to singular.

first i hit this:

```
kinds/preferences is not a valid kind: machineName: conflicting values "preference" and "preferences":
    /github.com/grafana/kindsys/kindcats.cue:46:15
    /github.com/grafana/kindsys/kindcats.cue:98:2
    /github.com/grafana/kindsys/kinds/constraint.cue:7:1
    /github.com/grafana/kindsys/kinds/preferences/preferences_kind.cue:7:20
```

and then this:
```
kinds/preferences: kind's machine name (Preference) must equal parent dir name (preferences)
exit status 1
kinds/gen.go:4: running "go": exit status 1
make: *** [Makefile:76: gen-cue] Error 1
```

**Comment by mildwonkey** [comment]
(summarizing our slack conversation): We shouldn't need to add any of these fields since they can be derived from the kind Name: https://github.com/grafana/kindsys/blob/ebfbbc0e58bf49a00a658341f3286ba5fecc056d/kindcats.cue#L46-L54; MachineName is used in several places in [the codegen templates](https://github.com/grafana/grafana/tree/main/pkg/codegen/tmpl). 

Is there a specific place we expected to see one of these fields, but didn't?

**Review by mildwonkey — APPROVED** [review]
LGTM (I'm always pro-explicit definitions), but pls double check that nothing is depending on those generated ts files. I didn't find anything, and we didn't break any integration tests, but I'm not the frontend expert (i'm not a frontend anything).

File: kinds/playlist/playlist_kind.cue [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 
-name:        "Playlist"
-maturity:    "merged"
-description: "A playlist is a series of dashboards that is automatically rotated in the browser, on a configurable interval."
+name:              "Playlist"
+maturity:          "merged"
+description:       "A playlist is a series of dashboards that is automatically rotated in the browser, on a configurable interval."
+pluralName:        "Playlists"
+machineName:       "playlist"
+pluralMachineName: "playlists"
```
Comment (sdboyer) [inline kinds/playlist/playlist_kind.cue:8]: What's the value of explicitly writing these in? All of these are the enforced (`machineName` and `pluralMachineName`) or default (`pluralName`) values, as specified by kindsys

File: kinds/preference/preference_kind.cue (hunk 1 above)
Comment (toddtreece) [reply kinds/preference/preference_kind.cue:3]: > by having singular and plural be the same (as indeed, they were before)

@sdboyer i reset the core kinds on #69115 and was able to get preferences to register. 

here's the default:

```
$ kubectl api-resources                                                 
NAME                         SHORTNAMES   APIVERSION                             NAMESPACED   KIND
preferencess                              preferences.kinds.grafana.com/v1       true         Preferences
```

it looks like `preferencess` is the default plural machine name

Comment (sdboyer) [reply kinds/preference/preference_kind.cue:3]: It is, but if you do:

```cue
name: "Preferences"
pluralName: "Preferences"
```

Then the plural machine name should also be the same as the singular machine name.

Comment (toddtreece) [reply kinds/preference/preference_kind.cue]: @sdboyer thanks, updated

**Review by sdboyer — APPROVED** [review]
LGTM!

**Review by ryantxu — APPROVED** [review]

**Pull request merged**